### PR TITLE
Make space for visually stacked branches 

### PR DESCRIPTION
**Pull request by brenetic**
We do not allow branches to be stacked in terms of data integrity but we
do allow them to be visually stacked. In these instances we need to make
space for branch expressions ('and' and 'or' types) so that the frontend
has the ability to display the question and answers in the pages flow.

At the same time this change also takes into account secondary routes
that may or may not link to pages and branches that have been previously
linked to earlier in the flow. In those instances we want to take the
later column, and under some circumstances we want to also update the
row numbers.

In order to accomplish this we have broken out the grid object into smaller
more easily testable models which wrap Coordinates, Rows and Columns.

Also add another branching fixture:

branching_10

Publish 2.11.0

### CODE REVIEW

File: app/models/metadata_presenter/coordinates.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+          return uuid
+        end
+      end
+    end
+
+    def position(uuid)
+      positions[uuid]
+    end
+
+    def occupied?(column, row, new_object_uuid)
```
Comment (tomas-stefano) [inline app/models/metadata_presenter/coordinates.rb:38]: Cool 👍

File: app/models/metadata_presenter/coordinates.rb [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    private
+
+    attr_reader :flow
+    attr_writer :positions
+
+    def setup_positions
+      flow.keys.index_with { |_uuid| { row: nil, column: nil } }
+    end
+  end
+end
```
Comment (tomas-stefano) [inline app/models/metadata_presenter/coordinates.rb:59]: Great code 🚀

**Review by tomas-stefano — APPROVED** [review]
🚀 🚢

**Review by njseeto — APPROVED** [review]
🏅 🚀 💯

**Pull request merged**